typescript-light-bulb - Daniel Yount

TheDanYount opened the pull request:
![Recording 2024-07-23 at 14 03 40](https://github.com/user-attachments/assets/c0f60d93-ab55-4856-bfb6-a360312fb4bd)

TheDanYount committed:
Completed the exercise

brandon-moy commented on typescript-light-bulb/styles.css:
```diff
@@ -0,0 +1,32 @@
+  border-radius: 125px;
+}
+
+.bulb-off {
+  background-color: rgb(25, 25, 25);
+  box-shadow: none;
+}
+
```
You could still give this a shadow for practice styling it. If you look closely at the example gif, there's actually a solid, dark shadow above the lightbulb.

brandon-moy commented on typescript-light-bulb/main.ts:
```diff
@@ -0,0 +1,18 @@
+const $bulb = document.querySelector('.bulb');
+const $background = document.querySelector('.background');
+let lightStatus = 'off';
+if (!$bulb) throw new Error('Someone had better hurry over to Home Depot!');
+$bulb.addEventListener('click', clickHandler);
```
Rather than using a `string` here, what is another data type we could use instead that would make more sense for two options?

brandon-moy reviewed (changes requested):
Great job, couple of spots I think you can improve this 👍

TheDanYount committed:
Small syntax and style fix

Made the binary choice of light a boolean, added a shadow for the lightbulb in the darkness

brandon-moy commented on typescript-light-bulb/main.ts:
```diff
@@ -0,0 +1,18 @@
+$bulb.addEventListener('click', clickHandler);
+function clickHandler(): void {
+  if (!$bulb) throw new Error('Someone had better hurry over to Home Depot!');
+  if (!$background) throw new Error("You're off the map!");
+  if (lightOn === false) {
+    lightOn = true;
+    $bulb.className = 'bulb bulb-on';
```
No change needed on this now, but you could can shorten this with _truthy_/_falsy_ evaluation. Instead of `if (lightOn === false)` you could just go with `if (!lightOn)` which evaluates if `lightOn` is not `true`.

brandon-moy reviewed (approved):
Great job! Left an additional comment for an additional minor improvement but not required. Approved! 👍
